api/filters/platform_termination_filter.go

```diff
@@ -62,11 +65,118 @@ func (f *platformTerminationFilter) Run(req *web.Request, next web.Handler) (*we
 				StatusCode:  http.StatusUnprocessableEntity,
 			}
 		}
-	}
+		instanceInOtherPlatforms, err := hasSharedInstancesInOtherPlatforms(ctx, platform, f.repository)
+		if err != nil {
+			return nil, err
+		}
 
+		if instanceInOtherPlatforms {
+			sharedInstancesReferences, err := findReferencesOfSharedInstancesInOtherPlatforms(ctx, platform, f.repository)
+			if err != nil {
+				return nil, err
+			}
+			return nil, &util.HTTPError{
+				ErrorType:   "UnprocessableEntity",
+				Description: "Platform cannot be deleted because other platform(s) has reference instance(s) to the shared instances in given platform. Details: " + formatSharingReferences(sharedInstancesReferences),
+				StatusCode:  http.StatusUnprocessableEntity,
+			}
+		}
+	}
 	return next.Handle(req)
 }
 
+func formatSharingReferences(references []*SharingReferences) string {
+	var msg []string
+	for i := 0; i < len(references); i++ {
+		msg = append(msg, "shared instance "+references[i].sharedInstanceID+" referenced by instance(s) "+strings.Join(references[i].referenceIDs, ", "))
+	}
+	return strings.Join(msg, ", ")
+}
+
+type SharingReferences struct {
+	sharedInstanceID string
+	referenceIDs     []string
+}
+
+func hasSharedInstancesInOtherPlatforms(ctx context.Context, platform *types.Platform, repository storage.Repository) (bool, error) {
+
+	sharedInstanceIDs, err := findSharedInstancesInPlatform(ctx, platform, repository)
+	if err != nil {
+		return false, err
+	}
+
+	if len(sharedInstanceIDs) == 0 {
+		return false, nil
+	}
+
+	references, err := repository.Count(ctx,
+		types.ServiceInstanceType,
+		query.ByField(query.InOperator, instance_sharing.ReferencedInstanceIDKey, sharedInstanceIDs...),
+		query.ByField(query.NotEqualsOperator, "platform_id", platform.GetID()))
+
+	if err != nil {
+		return false, util.HandleStorageError(err, types.ServiceInstanceType.String())
+	}
+
+	if references > 0 {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func findReferencesOfSharedInstancesInOtherPlatforms(ctx context.Context, platform *types.Platform, repository storage.Repository) ([]*SharingReferences, error) {
```

Comment:
wouldn't it be simpler to just return the instances instead of the abstraction with SharingReference? what is the value we get from this type?

Reply:
this type is used with formatSharingReferences, there the shared instance ID + all it's references are interesting

Reply:
yes, I saw that, and wasn't sure if that specific error message format is worth all the fuss :)

Reply:
10x 👍 
changed it to return instances & moved the format part to formatSharingReferences